libnavigation-core/src/test/java/com/mapbox/navigation/core/internal/HistoryRecordingStateHandlerTest.kt

```diff
@@ -739,13 +752,65 @@ class HistoryRecordingStateHandlerTest {
 
             override fun onShouldCancelRecording(state: NavigationSessionState) { }
         }
-        stateHandler = HistoryRecordingStateHandler(NavigationSessionState.Idle)
+        stateHandler = HistoryRecordingStateHandler(coroutineRule.coroutineScope)
         stateHandler.registerStateChangeObserver(unregisteringObserver)
         stateHandler.registerStateChangeObserver(observer)
         stateHandler.onSessionStateChanged(TripSessionState.STARTED)
         // no ConcurrentModificationException
     }
 
+    @Test
+    fun flowUpdates() = coroutineRule.runBlockingTest {
```

Comment:
`returnsMany` should help ti reduce complexity here. ref https://mockk.io/

Reply:
I'm not sure it would reduce it. The way it is I mock the method with a desired answer just before the invocation that will use this answer. If I set up `NavigationSessionUtils#getNewState` answers in one place it will be harder to track why the nth answer should be exactly like this. And if I add or remove a step, I'd have to change this set up as well and it might be easy to forget. That's if I correctly understood your suggestion.